tests/Unit/CredentialManagersTest.php

```diff
@@ -178,6 +185,36 @@ public function test_you_can_get_user($sutClass, $setupFunction, $createExisting
         ], $sut->getUser());
     }
 
+    #[DataProvider('credentialManagers')]
+    public function test_that_getting_a_non_existent_tenant_will_throw_an_exception($sutClass, $setupFunction, $createExistingData)
+    {
+        $setupFunction();
+
+        $sut = new $sutClass();
+
+        $createExistingData($sut, [
+            'tenants' => []
+        ]);
+
+        $this->assertThrows(fn () => $sut->getTenantId(1), XeroTenantNotFound::class, 'No such tenant exists');
+    }
+
+    public function test_that_if_guest_it_will_throw_exception_for_authenticated_user_store()
+    {
+        $this->assertThrows(fn () => new AuthenticatedUserStore(), XeroUserNotAuthenticated::class, 'User is not authenticated');
```

Comment:
I've added a test for this, but it feels wrong to throw this exception in the constructor.

It might be better placed in the `data()` method. 

Thoughts?

Reply:
Yeah, I think `data()` is the right call, that way it will attempt to resolve the user as when we try and actually use the Credential Manager, rather that just from building it.